### PR TITLE
Stm32mp15 disco flashlayout

### PR DESCRIPTION
**Pull request by igoropaniuk**

### CODE REVIEW

File: meta-lmp-bsp/dynamic-layers/stm-st-stm32mp/recipes-support/stm32-mfgtool-files/stm32-mfgtool-files/provision.sh.in [hunk 1]
```diff
@@ -344,7 +344,7 @@ if [ "${need_reboot}" -eq "1" ]; then
 	fastboot reboot
 	sleep 2
 	# use TSV with usb entries only, as we don't want to re-flash binaries to eMMC again
-	STM32_Programmer_CLI -c port="${hwPort}" -d stm32-mfgtool-files/FlashLayout_emmc_@@BOARD_NAME@@-optee.tsv
+	STM32_Programmer_CLI -c port="${hwPort}" -d stm32-mfgtool-files/FlashLayout_stm32mp1-usb.tsv
```
Comment (MrCry0) [inline meta-lmp-bsp/dynamic-layers/stm-st-stm32mp/recipes-support/stm32-mfgtool-files/stm32-mfgtool-files/provision.sh.in:347]: Could we avoid using the same literal in `lmp-mfgtool-machine-custom.inc` and in `provision.sh.in`?

like:
```
STM32MP1_USB_FLASHLAYOUT = "FlashLayout_stm32mp1-usb.tsv"
LMP_FLASHLAYOUT:stm32mp1common = ${STM32MP1_USB_FLASHLAYOUT}
...
STM32_Programmer_CLI -c port="${hwPort}" -d stm32-mfgtool-files/@@FLASHLAYOUT_USB@@
...
sed ...
```

Comment (igoropaniuk) [reply meta-lmp-bsp/dynamic-layers/stm-st-stm32mp/recipes-support/stm32-mfgtool-files/stm32-mfgtool-files/provision.sh.in]: fixed, thanks!

File: meta-lmp-bsp/wic/sdimage-stm32mp157c-dk2-optee-sota.wks.in [hunk 2]
```diff
@@ -15,7 +15,7 @@ part fsbl2 --source rawcopy --fstype=ext4 --fsoptions "noauto" --part-name=fsbl2
 part fip1 --source rawcopy --fstype=ext4 --fsoptions "noauto" --part-name=fip-a --sourceparams="file=${DEPLOY_DIR_IMAGE}/fip/fip-stm32mp157c-dk2-optee.bin" --ondisk mmcblk --part-type 0x8301 --fixed-size 4096K
 part fip2 --source rawcopy --fstype=ext4 --fsoptions "noauto" --part-name=fip-b --sourceparams="file=${DEPLOY_DIR_IMAGE}/fip/fip-stm32mp157c-dk2-optee.bin" --ondisk mmcblk --part-type 0x8301 --fixed-size 4096K
 
-part u-boot-env --source empty --part-name=uboot-env --ondisk mmcblk --part-type 0x8301 --fixed-size 16K --align 8192
+part u-boot-env --source empty --part-name=uboot-env --ondisk mmcblk --part-type 0x8301 --fixed-size 16K --align 9216
```
Comment (MrCry0) [inline meta-lmp-bsp/wic/sdimage-stm32mp157c-dk2-optee-sota.wks.in:18]: I would also change the layout scheme above (lines 4-11).

With or without this, LGTM

File: meta-lmp-bsp/wic/sdimage-stm32mp157c-dk2-optee.wks.in [hunk 3]
```diff
@@ -15,7 +15,7 @@ part fsbl2 --source rawcopy --fstype=ext4 --fsoptions "noauto" --part-name=fsbl2
 part fip1 --source rawcopy --fstype=ext4 --fsoptions "noauto" --part-name=fip-a --sourceparams="file=${DEPLOY_DIR_IMAGE}/fip/fip-stm32mp157c-dk2-optee.bin" --ondisk mmcblk --part-type 0x8301 --fixed-size 4096K
 part fip2 --source rawcopy --fstype=ext4 --fsoptions "noauto" --part-name=fip-b --sourceparams="file=${DEPLOY_DIR_IMAGE}/fip/fip-stm32mp157c-dk2-optee.bin" --ondisk mmcblk --part-type 0x8301 --fixed-size 4096K
 
-part u-boot-env --source empty --part-name=uboot-env --ondisk mmcblk --part-type 0x8301 --fixed-size 16K --align 8192
+part u-boot-env --source empty --part-name=uboot-env --ondisk mmcblk --part-type 0x8301 --fixed-size 16K --align 9216
```
Comment (MrCry0) [inline meta-lmp-bsp/wic/sdimage-stm32mp157c-dk2-optee.wks.in:18]: The same.

**Review by MrCry0 — APPROVED** [review]

**Review by quaresmajose — APPROVED** [review]
LGTM

**Review by ricardosalveti — APPROVED** [review]
LGTM

**Pull request merged**